src/main/java/DecoratorForString.java

```diff
@@ -0,0 +1,46 @@
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class DecoratorForString implements Links{
```

Comment:
@PavelNejinskiy implements `Links<String>` - we want to emphasize that we are implementing String links interface. Also, the name of this class and URL based one not quite accurate. What are we decorating here?

The general rule for class naming is to use main interface implemented name + specifics describing the implementation, like `StringLinks`